### PR TITLE
test: [M3-8444] - Add assertions for bucket details drawer tests

### PR DESCRIPTION
**Pull request by AzureLatte**
## Description 📝
Add a Cypress integration test to cover OBJ bucket details drawer changes for OBJ Gen 2. 

## Changes  🔄
- Add assertions for bucket details drawer tests
- Add new util function `mockGetBucketAccess`

## How to test 🧪
```
yarn cy:run -s "cypress/e2e/core/objectStorageGen2/bucket-create-gen2.spec.ts"
```

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/cypress/e2e/core/objectStorageGen2/bucket-create-gen2.spec.ts [hunk 1]
```diff
@@ -102,14 +103,72 @@ describe('Object Storage Gen2 create bucket tests', () => {
...
+      ) {
+        cy.contains(bucketRateLimitsNotice).should('be.visible');
+        cy.get('[data-testid="bucket-rate-limit-table"]').should('be.visible');
+        cy.contains(CORSNotice).should('be.visible');
+      } else {
+        cy.get('[data-testid="bucket-rate-limit-table"]').should('not.exist');
+        ui.toggle
+          .find()
+          .should('have.attr', 'data-qa-toggle', 'true')
+          .should('be.visible');
```
Comment (mjac0bs) [inline packages/manager/cypress/e2e/core/objectStorageGen2/bucket-create-gen2.spec.ts:130]: Did we need to check that this toggle is *not* visible for E2 and E3?

**Review by mjac0bs — COMMENTED** [review]
I ran these tests locally on an account that didn't have Object Storage enabled in my Settings, and the tests failed consistently at the point where the 'Enable Object Storage' modal opens after Create Button is clicked. Should this test spec account for that?

Once I enabled OBJ, tests all passed and cover the cases outlined in the internal ticket. ✅

**Review by cpathipa — COMMENTED** [review]
I think we have three variations for OBJ feature legacy, gen1 and gen2 are we planing to cover those scenarios with mock flags ?

File: packages/manager/cypress/e2e/core/objectStorageGen2/bucket-create-gen2.spec.ts (hunk 1 above)
Comment (AzureLatte) [reply packages/manager/cypress/e2e/core/objectStorageGen2/bucket-create-gen2.spec.ts:131]: Updated, thanks!

File: packages/manager/cypress/support/intercepts/object-storage.ts [hunk 2]
```diff
@@ -504,3 +504,26 @@ export const mockGetObjectStorageEndpoints = (
...
+  cluster: string
+): Cypress.Chainable<null> => {
+  return cy.intercept(
+    'GET',
+    apiMatcher(`object-storage/buckets/${cluster}/${label}/access`),
+    {
+      body: {},
+      statusCode: 200,
+    }
+  );
```
Comment (coliu-akamai) [inline packages/manager/cypress/support/intercepts/object-storage.ts:528]: Hey @AzureLatte just a heads up that I'd also created a mock intercept for bucket access [here](https://github.com/linode/manager/pull/10994/files#diff-d695c95822eca6ee863752b6c1c1a06d4f200173d9eef2ea7fadee757dcbe087). Will keep an eye on this PR, and if this gets merged in first will update my PR accordingly!

**Comment by mjac0bs** [comment]
@AzureLatte This one's got a couple merge conflicts now to be resolved.

Screenshot 2024-10-09 at 9 40 38 AM

**Review by coliu-akamai — APPROVED** [review]
✅ confirmed tests passed - thanks Azure!

Since this file is named `bucket-create`, wondering if we should update the name to something like `bucket-create-and-details-drawer` or something (maybe that's too wordy), or move these assertions to a different file?

File: packages/manager/cypress/e2e/core/objectStorageGen2/bucket-create-gen2.spec.ts [hunk 3]
```diff
@@ -71,14 +72,80 @@ describe('Object Storage Gen2 create bucket tests', () => {
...
+      // Close the Details drawer
+      cy.get('[data-qa-close-drawer="true"]').should('be.visible').click();
+    });
+  };
+
   /**
    * Confirms UI flow for creating a gen2 Object Storage bucket with endpoint E0
    * Confirms all endpoints are displayed regardless if there's multiple of the same type
    * Confirms S3 endpoint hostname displayed to differentiate between identical options in the dropdown
+   * Confirms correct information displays in the details drawer for all endpoint types
```
[the remark this answers is not in the document]
Comment (AzureLatte) [reply packages/manager/cypress/e2e/core/objectStorageGen2/bucket-create-gen2.spec.ts:143]: Nice suggestion, I'll update it, thank you!
